.github/workflows/security-scan.yml

```
@@ -0,0 +1,173 @@
+      - name: Run ESLint Security Analysis
+        run: |
+          npx eslint . --ext .js,.ts --format json > reports/eslint-security.json || echo "ESLint security scan completed"
+
+      - name: Run Custom Security Scanner
+        run: |
+          chmod +x security-scan.sh
+          ./security-scan.sh
+
+      - name: Check for Critical Issues
+        id: security-check
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2086:info:1:42: Double quote to prevent globbing and word splitting

(shellcheck)